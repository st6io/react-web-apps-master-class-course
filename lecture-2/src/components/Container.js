import React from 'react';
import styles from './common.module.scss';

const Container = ({ className, children }) => (
  <div className={`${styles.container} ${className || ''}`}>{children}</div>
);

export default Container;
