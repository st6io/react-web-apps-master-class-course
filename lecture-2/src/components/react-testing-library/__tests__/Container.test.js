import React from 'react';
import { render } from '@testing-library/react';

import Container from '../../Container';

describe('Container Component', () => {
  it('should match snapshot', () => {
    const { container } = render(<Container />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with className', () => {
    const { container } = render(<Container className={'new-class-name'} />);

    expect(container).toMatchInlineSnapshot(`
      <div>
        <div
          class="container new-class-name"
        />
      </div>
    `);
  });
});
