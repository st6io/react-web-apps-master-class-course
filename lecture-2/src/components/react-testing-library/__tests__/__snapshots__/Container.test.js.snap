// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component should match snapshot 1`] = `
<div>
  <div
    class="container "
  />
</div>
`;
